<?xml version="1.0"?>

<!-- Firefly test launch file -->
<launch>
	
	<param name="use_sim_time" value="True"/>	
	
    <node name="c_localization" 
          pkg="c_slam_roamfree" 
          type="c_localization"  
          output="screen" 
          launch-prefix="">
    
     	<!-- Ros stuff -->
		<param name="trackedFrame" value="camera_link" />
		<param name="imuTopic" value="/firefly/ground_truth/imu" />

		<!-- Initial pose -->
		<rosparam param="initialPose">[0.0, 0.0, 0.08, 0.5, -0.5, 0.5, -0.5]</rosparam>

		<rosparam param="T_O_IMU">[0.0, 0.0, 0.0, 0.5, 0.5, -0.5, 0.5]</rosparam>

		<rosparam param="T_O_CAMERA">[0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0]</rosparam>

		<!-- Imu bias -->
		<param name="isAccBiasFixed" value="True" />
		<rosparam param="accBias">[0.0, 0.0, 0.0]</rosparam>
		
		<param name="isGyroBiasFixed" value="True" />
		<rosparam param="gyroBias">[0.0, 0.0, 0.0]</rosparam>

		<!-- Imu data -->
		<param name="imu_N" value="10" />
		<param name="imu_dt" value="0.01" />

		<!-- Camera calibration -->
		<rosparam param="K">[460.890258789062, 0.0, 343.462058902507, 0.0, 529.505432128906, 201.160554159542, 0.0, 0.0, 1.0]</rosparam>

		<!-- Vision Stuff -->
		<param name="initialScale" value="5.0" />
		<param name="minWindowLenghtSecond" value="2.0" />
		<param name="minActiveFeatures" value="3" />

		<!-- Optimization method -->
		<param name="useGaussNewton" value="True" />
		<param name="iterationN" value="1" />
    </node>
          
	 <node name="firefly_room_test" 
          pkg="c_slam_roamfree" 
          type="firefly_room_test" 
          output="screen" 
          launch-prefix=""/>
          
          <node name="displayer" 
          pkg="c_slam_roamfree" 
          type="displayer" 
          output="screen" 
          launch-prefix=""/>

    <node name="rviz" 
          pkg="rviz" 
          type="rviz"
          respawn="true"
          args="-d $(find c_slam_roamfree)/rviz_config/rviz.rviz"/>
</launch>